packages/core/src/utils/memoryDiscovery.ts

```diff
@@ -29,7 +29,8 @@ const logger = {
     console.error('[ERROR] [MemoryDiscovery]', ...args),
 };
 
-const MAX_DIRECTORIES_TO_SCAN_FOR_MEMORY = 200;
+const MAX_DIRECTORIES_TO_SCAN_FOR_MEMORY =
+  parseInt(process.env.GEMINI_MEMORY_DISCOVERY_MAX_DIRS ?? '', 10) || 200;
 
```

Comment:
it would make this a bit more annoying to use as settings.json is in the CLI package but what do you think of making this be in settings.ts rather than an env variable? this kindof seems like something I might want to set in settings.json. For example, maybe my project is really large so I want a specific value for it.
in general I'm trying to keep most settings in settings.ts where feasible rather than encouraging more env variables or options in config controlled by args unless they strictly have to go that way. Happy to be convinced otherwise for this case.